pallets/regulated-assets/src/lib.rs

```diff
@@ -0,0 +1,305 @@
+                <Error<T>>::OnlyAssetOwnerCanRegulate
+            );
+            ensure!(
+                !Self::regulated_asset(asset_id),
+                <Error<T>>::AssetAlreadyRegulated
+            );
+
+            <RegulatedAsset<T>>::set(asset_id, true);
+            Self::deposit_event(Event::AssetRegulated { asset_id });
+
+            Ok(())
+        }
+
+        /// Issues a new Soulbound Token (SBT).
+        ///
+        /// ## Parameters
+        ///
+        /// - `origin`: The origin of the transaction.
+        /// - `symbol`: The symbol of the SBT which should represent string with only uppercase latin chars with max length of 7.
+        /// - `name`: The name of the SBT should represent string with only uppercase or lowercase latin chars or numbers or spaces, with max length of 33.
+        /// - `allowed_assets`: TThe list of assets allowed to be operated with by holding the SBT.
+        /// - `description`: The description of the SBT. (Optional)
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::issue_sbt())]
+        pub fn issue_sbt(
+            origin: OriginFor<T>,
+            symbol: AssetSymbol,
+            name: AssetName,
+            allowed_assets: Vec<AssetIdOf<T>>,
+            description: Option<Description>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // Check permission `who` can issue SBT
+            Permissions::<T>::check_permission(who.clone(), ISSUE_SBT)?;
+
+            let sbt_asset_id = T::AssetManager::register_from(
+                &who,
+                symbol,
+                name.clone(),
+                DEFAULT_BALANCE_PRECISION,
+                0,
+                true,
+                None,
+                description.clone(),
+            )?;
+            let metadata = SoulboundTokenMetadata {
+                name,
+                description,
+                allowed_assets: allowed_assets.clone(),
+            };
+            <SoulboundAsset<T>>::insert(sbt_asset_id, &metadata);
+
+            for allowed_asset in allowed_assets {
+                <SBTsByAsset<T>>::mutate(allowed_asset, |sbts| {
+                    sbts.insert(sbt_asset_id);
+                });
+            }
+
+            Self::deposit_event(Event::SoulboundTokenIssued {
+                asset_id: sbt_asset_id,
+                owner: who,
+                metadata,
+            });
+
+            Ok(())
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Emits When an asset is regulated
+        AssetRegulated { asset_id: AssetIdOf<T> },
+        /// Emits When an SBT is issued
+        SoulboundTokenIssued {
+            asset_id: AssetIdOf<T>,
+            owner: AccountIdOf<T>,
+            metadata: SoulboundTokenMetadata<AssetIdOf<T>>,
+        },
```

Comment:
Probably we have to add `allowed_assets`, what do you think?

Reply:
I think you're right. 
I will add it with other attributes in the next task as the SBT support is the main part of the next [milestone](https://github.com/sora-xor/sora2-network/issues/1013)
for this task I partially supported SBT to run against the cases of regulated assets' operation

@wer1st

Reply:
It is just add one line here. It's too easy to forget it in the next task.
If you add the event in this task, you have change it in the scope of this task. Don't postpone it